### PR TITLE
sfs turbo module

### PR DESCRIPTION
**Pull request by Polina-Gubina**

### CODE REVIEW

File: plugins/modules/sfsturbo_share_action.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+   timeout:
+     description:
+       - Specifies the timeout.
+     type: int
+     default: 350
+requirements: ["openstacksdk", "otcextensions"]
+'''
+
+EXAMPLES = '''
+- name: Extend capacity vpc
```
Comment (vladimirvshivkov) [inline plugins/modules/sfsturbo_share_action.py:50]: not relevant to sfs

File: plugins/modules/sfsturbo_share_action.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+  opentelekomcloud.cloud.sfsturbo_share_action:
+    name: "share-test"
+    new_size: 200
+
+- name: Change security group
```
Comment (vladimirvshivkov) [inline plugins/modules/sfsturbo_share_action.py:55]: also here

File: plugins/modules/sfsturbo_share_action.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+DOCUMENTATION = '''
+---
+module: sfsturbo_share_action
```
Comment (vladimirvshivkov) [inline plugins/modules/sfsturbo_share_action.py:16]: not clear what this module does

File: meta/runtime.yml [hunk 4]
```diff
@@ -58,6 +58,7 @@ action_groups:
     - server_group_info
     - subnet
     - subnet_info
+    - sfsturbo
```
Comment (vladimirvshivkov) [inline meta/runtime.yml:61]: here should be listed all new modules

File: tests/integration/targets/sfsturbo/aliases [hunk 5]
```diff
@@ -0,0 +1,2 @@
+disabled
+sfsturbo/group1
```
Comment (vladimirvshivkov) [inline tests/integration/targets/sfsturbo/aliases:2]: new line needed

**Review by vladimirvshivkov — CHANGES_REQUESTED** [review]

**Review by artem-lifshits — APPROVED** [review]

File: plugins/modules/sfsturbo_share_action.py [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+short_description: Extend capacity and security group change for share
+extends_documentation_fragment: opentelekomcloud.cloud.otc
+author: "Polina Gubina (@polina-gubina)"
+version_added: "0.10.0"
```
Comment (anton-sidelnikov) [inline plugins/modules/sfsturbo_share_action.py:20]: again should be next from latest 0.15.0

File: plugins/modules/sfsturbo_share_action.py [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+        required_if=[
+            ('id', None, ['name']),
+            ('name', None, ['id'])
+        ],
+        supports_check_mode=True
+    )
+
+    def build_update(self, share, security_group_id):
+        if self.params['new_size']:
+            if int(float(share.avail_capacity)) != self.params['new_size']:
```
Comment (anton-sidelnikov) [inline plugins/modules/sfsturbo_share_action.py:98]: why float here? and then int conversion, loooks strange

Comment (anton-sidelnikov) [reply plugins/modules/sfsturbo_share_action.py:98]: share.avail_capacity is a string  or what?

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]

File: plugins/modules/sfsturbo_share.py [hunk 8]
```diff
@@ -147,6 +164,49 @@ def _create(self, attributes, timeout, wait, **kwargs):
...
+
+    def _build_update(self, resource, attributes, updateable_attributes,
+                      non_updateable_attributes, **kwargs):
+        update = {}
+        resource = self.find_function(name_or_id=resource['id'])
+
+        resource_attributes = {}
+
+        if attributes.get('size'):
+            if attributes['size'] != int(float(resource['size'])):
```
Comment (anton-sidelnikov) [inline plugins/modules/sfsturbo_share.py:197]: check please on bigger value

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]

**Review by anton-sidelnikov — APPROVED** [review]

**Review by anton-sidelnikov — APPROVED** [review]

**Review by artem-lifshits — APPROVED** [review]

**Review by vladimirvshivkov — APPROVED** [review]

**Pull request merged**